src/components/ClassInfoModal.tsx

```diff
@@ -133,41 +136,82 @@ export const ClassInfoModal: React.FC<ClassInfoModalProps> = ({
                 </ModalBody>
 
                 <ModalFooter>
-                    <Link
-                        href={
-                            me
-                                ? me.role === roles.VOLUNTEER
-                                    ? `/volunteer/enrollment?classId=${classInfo.id}`
-                                    : `/parent/enrollment/?classId=${classInfo.id}`
-                                : "/login"
-                        }
-                    >
-                        <Button
-                            bg={colourTheme.colors.Blue}
-                            color={"white"}
-                            mx={"auto"}
-                            my={2}
-                            fontWeight={"200"}
-                            _hover={{
-                                textDecoration: "none",
-                                bg: colourTheme.colors.LightBlue,
-                            }}
-                            _active={{
-                                bg: "lightgrey",
-                                outlineColor: "grey",
-                                border: "grey",
-                                boxShadow: "lightgrey",
-                            }}
-                            _focus={{
-                                outlineColor: "grey",
-                                border: "grey",
-                                boxShadow: "lightgrey",
-                            }}
-                            minW={"100%"}
+                    {isFull ? (
+                        <ModalBody>
+                            <Button
+                                bg={colourTheme.colors.Blue}
+                                color={"white"}
+                                mx={"auto"}
+                                my={2}
+                                fontWeight={"200"}
+                                _hover={{
+                                    textDecoration: "none",
+                                    bg: colourTheme.colors.LightBlue,
+                                }}
+                                _active={{
+                                    bg: "lightgrey",
+                                    outlineColor: "grey",
+                                    border: "grey",
+                                    boxShadow: "lightgrey",
+                                }}
+                                _focus={{
+                                    outlineColor: "grey",
+                                    border: "grey",
+                                    boxShadow: "lightgrey",
+                                }}
+                                minW={"100%"}
+                                onClick={() =>
+                                    createWaitlistRegistration(
+                                        me.parent,
+                                        classInfo.id,
+                                    )
+                                }
```

Comment:
```suggestion
                                onClick={() => {
                                    createWaitlistRegistration(
                                        me.parent,
                                        classInfo.id,
                                    );
                                    toast({
                                        title: "Class waitlisted.",
                                        description:
                                            "We'll notify you once space becomes available",
                                        status: "success",
                                        duration: 9000,
                                        isClosable: true,
                                        position: "top-right",
                                    });
                                    onClose();
                                }}
```
let's close the modal after we are done with it. We can also get `toast = useToast()` from chakra ui and use that to indicate a successful waitlist.

https://user-images.githubusercontent.com/44826218/140004815-e8e14ff7-879c-4b89-924d-ca4b0b02dfae.mp4

@vedantdsn @bekahma let us know if this is okay.